src/box/memtx_engine.c

```diff
@@ -1049,10 +1053,220 @@ memtx_engine_gc_f(va_list va)
 	return 0;
 }
 
+void
+memtx_engine_schedule_gc(struct memtx_engine *memtx,
+			 struct memtx_gc_task *task)
+{
+	stailq_add_tail_entry(&memtx->gc_queue, task, link);
+	fiber_wakeup(memtx->gc_fiber);
+}
+
+void
+memtx_engine_set_snap_io_rate_limit(struct memtx_engine *memtx, double limit)
+{
+	memtx->snap_io_rate_limit = limit * 1024 * 1024;
+}
+
+int
+memtx_engine_set_memory(struct memtx_engine *memtx, size_t size)
+{
+	if (size < quota_total(&memtx->quota)) {
+		diag_set(ClientError, ER_CFG, "memtx_memory",
+			 "cannot decrease memory size at runtime");
+		return -1;
+	}
+	quota_set(&memtx->quota, size);
+	return 0;
+}
+
+void
+memtx_engine_set_max_tuple_size(struct memtx_engine *memtx, size_t max_size)
+{
+	memtx->max_tuple_size = max_size;
+}
+
+void
+memtx_enter_delayed_free_mode(struct memtx_engine *memtx)
+{
+	memtx->snapshot_version++;
+	if (memtx->delayed_free_mode++ == 0)
+		memtx_allocator_enter_delayed_free_mode(memtx);
+}
+
+void
+memtx_leave_delayed_free_mode(struct memtx_engine *memtx)
```

Comment:
These changes look totally unrelated to the an allocator type. Please, extract such non-functional things to separate commits.

Reply:
Fixed